context/src/main/java/io/micronaut/scheduling/processor/ScheduledMethodProcessor.java

```diff
@@ -94,6 +94,13 @@ public void process(BeanDefinition<?> beanDefinition, ExecutableMethod<?, ?> met
             return;
         }
 
+        boolean containsInContext = beanContext.getBeansOfType(beanDefinition.getBeanType())
+            .stream()
+            .anyMatch(b -> b.getClass().equals(beanDefinition.getBeanType()));
+        if (!containsInContext) {
```

Comment:
not sure this is the correct fix. You can use `beanContext.containsBean(..)`.. you will need to include the qualifier

Reply:
Thanks for the feedback! I changed this code to use a qualifier that checks for an exact type match. Please take another look

Reply:
Thanks @codewithbear when you want another review click the 

![CleanShot 2023-09-22 at 09 31 27@2x](https://github.com/micronaut-projects/micronaut-core/assets/864788/d4ab4753-9aa4-49bf-b6a1-5adca4456a4c)


that will trigger another notification to the reviewer.